src/clj/orchestra/spec/test.cljc

```diff
@@ -124,13 +124,13 @@ failure in instrument."
       (if *instrument-enabled*
         (let [cargs (when-let [spec (:args fn-spec)]
                       (conform! v :args spec args ::s/args))
-              ret (.applyTo ^clojure.lang.IFn f args)]
-          (when-let [spec (:ret fn-spec)]
-            (conform! v :ret spec ret ::s/ret))
+              ret (.applyTo ^clojure.lang.IFn f args)
+              cret (when-let [spec (:ret fn-spec)]
+                     (conform! v :ret spec ret ::s/ret))]
           (when-let [spec (:fn fn-spec)]
             (if (nil? cargs)
               (throw (no-args-spec v fn-spec))
-              (conform! v :fn spec {:ret ret :args cargs} ::s/fn)))
+              (conform! v :fn spec {:ret cret :args cargs} ::s/fn)))
```

Comment:
I believe this logic has an issue. In the case where no `:fn` is provided (and one is certainly not required), the `:ret` value will no longer be taken from the fn application. Instead, it will be `nil` because of the `when-let`.

We probably want something like this:

```clojure
(let [; ...
      ret (.applyTo ^clojure.lang.IFn f args)
      cret (if-some [spec (:ret fn-spec)]
             (conform! v :ret spec ret ::s/ret)
             ret)]
  ; ...
  )
```